scripts/common/hardhat.utils.ts

```diff
@@ -339,4 +370,12 @@ export class HardhatUtils {
         const balanceAfter = await token.balanceOf(account)
         return balance == balanceAfter
     }
+    public async replaceImmutableAddress(toReplace: string, replacement: string, contract: Contract) {
+        const bytecode = await this.hre.ethers.provider.getCode(contract.address)
+        const bytecodeWithReplacement = bytecode.replaceAll(
```

Comment:
how do you know that are all ocurancess of an address ?

Reply:
This is only for immutable ones that are stored in runtime bytecode, should work the same as during compilation where al occurances in the code are replaced by the immutable address.